### PR TITLE
refactor: charge command orchestration

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

This is the first in a series of two or more pull requests with the purpose to improve the orchestration of charge command handling when updating charge state.

Due to changes in the Charge rejection flow, this pull request also introduces similar orchestration of charge links commands.

## References

* #1306

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeLinksDataAvailableNotifiedEvents/ChargeLinksDataAvailableNotifiedEventFactory.cs [hunk 1]
```diff
@@ -20,9 +21,13 @@ public class ChargeLinksDataAvailableNotifiedEventFactory : IChargeLinksDataAvai
     {
         public ChargeLinksDataAvailableNotifiedEvent Create(ChargeLinksAcceptedEvent chargeLinksAcceptedEvent)
         {
+            var meteringPointId = chargeLinksAcceptedEvent.ChargeLinksCommand.ChargeLinksOperations
+                .Single()
```
Comment (jonasdmoeller) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeLinksDataAvailableNotifiedEvents/ChargeLinksDataAvailableNotifiedEventFactory.cs:25]: If operations only have a single element, why is it a list?

**Review by jonasdmoeller — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/Validation/IBusinessValidationRulesFactory.cs [hunk 2]
```diff
@@ -17,9 +17,9 @@
 
 namespace GreenEnergyHub.Charges.Domain.Dtos.Validation
 {
-    public interface IBusinessValidationRulesFactory<in TCommand>
-        where TCommand : CommandBase
+    public interface IBusinessValidationRulesFactory<in TOperation>
+        where TOperation : OperationBase
     {
-        Task<IValidationRuleSet> CreateRulesAsync(TCommand command);
+        Task<IValidationRuleSet> CreateRulesAsync(TOperation command);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/Validation/IBusinessValidationRulesFactory.cs:23]:
```suggestion
        Task<IValidationRuleSet> CreateRulesAsync(TOperation operation);
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/Validation/IInputValidationRulesFactory.cs [hunk 3]
```diff
@@ -16,9 +16,9 @@
 
 namespace GreenEnergyHub.Charges.Domain.Dtos.Validation
 {
-    public interface IInputValidationRulesFactory<in TCommand>
-        where TCommand : CommandBase
+    public interface IInputValidationRulesFactory<in TOperation>
+        where TOperation : OperationBase
     {
-        IValidationRuleSet CreateRulesForCommand(TCommand command);
+        IValidationRuleSet CreateRulesForOperation(TOperation command);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/Validation/IInputValidationRulesFactory.cs:22]:
```suggestion
        IValidationRuleSet CreateRulesForOperation(TOperation operation);
```

**Review by prtandrup — COMMENTED** [review]
50%+ done, a few minor things so far.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ChargeCommandInputValidationRulesFactoryTests.cs [hunk 4]
```diff
@@ -31,14 +32,33 @@ namespace GreenEnergyHub.Charges.Tests.Domain.Dtos.ChargeCommands.Validation.Inp
     [UnitTest]
     public class ChargeCommandInputValidationRulesFactoryTests
     {
+        [Fact]
+        public void CreateRulesForChargeCommand_ShouldContainRulesTest()
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ChargeCommandInputValidationRulesFactoryTests.cs:36]: ChargeCommand versus ChargeOperation?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ChargeCommandInputValidatorTests.cs [hunk 5]
```diff
@@ -28,13 +28,13 @@ public class ChargeCommandInputValidatorTests
         [InlineAutoData]
         public void Validate_WhenValidatingChargeCommand_ReturnsChargeCommandValidationResult(
             ChargeCommandInputValidationRulesFactory chargeCommandInputValidationRulesFactory,
-            ChargeCommand chargeCommand)
+            ChargeOperationDto chargeOperationDto)
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Dtos/ChargeCommands/Validation/InputValidation/ChargeCommandInputValidatorTests.cs:31]: ChargeCommand versus ChargeOperation?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Validation/BusinessValidatorTests.cs [hunk 6]
```diff
@@ -29,15 +30,15 @@ public class BusinessValidatorTests
         [Theory]
         [InlineAutoMoqData]
         public async Task ValidateAsync_WhenCalled_WithCommand_UsesFactoryToFetchRulesAndUseRulesToGetResult(
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Domain/Validation/BusinessValidatorTests.cs:32]:
```suggestion
        public async Task ValidateAsync_WhenCalled_WithOperation_UsesFactoryToFetchRulesAndUseRulesToGetResult(
```

**Review by prtandrup — COMMENTED** [review]
Review done, good job with the refactoring. I've added a few comments. Majority about consistency; ChargeCommand versus ChargeOperationDto.

**Pull request merged**